automation/taskcluster/decision_task.py

```diff
@@ -76,16 +76,89 @@ def create_raw_task(name, description, full_command, scopes = []):
+                    "GeckoReleaseAarch64", "GeckoReleaseArm", "GeckoReleaseX86"
+                ]
+            ),
+            "lintTask": "lintGeckoBetaArmDebug"
+        },
+        ":support-test": {
+            "lintTask": "lint"
+        },
+        ":tooling-lint": {
+            "lintTask": "lint"
+        }
+    }
 
+    def make_task(gradle_task, subtitle=""):
+        return create_task(
+            name="Android Components - Module " + module + " " + subtitle,
+            description='Running configured tasks for ' + module,
+            command="-Pcoverage " + gradle_task + " && automation/taskcluster/action/upload_coverage_report.sh",
+            scopes = [
+                "secrets:get:project/mobile/android-components/public-tokens"
+            ])
+
+    def as_module_task(module, task):
+        return module + ":" + task
+
+    tasks = []
+
+    override = module_overrides.get(module, {})
+
+    if "assembleOnly/Test" in override:
+        # If assemble/test variant lists are specified, create individual tasks as appropriate,
+        # while also checking if we can sneak in a specified lint task along with a matching assemble
+        # or test variant. If lint task isn't specified
+
+        assemble_only, assemble_and_test = override["assembleOnly/Test"]
+        lint_task = override.get("lintTask", "")
+        scheduled_lint = False
+
+        # Schedule "assemble only" tasks; if a lintTask matches any variants, schedule it as well.
+        for variant in assemble_only:
+            task = as_module_task(module, "assemble" + variant)
+            subtitle = "assemble"
+            if variant in lint_task:
+                task = task + " " + as_module_task(module, lint_task)
+                subtitle = subtitle + "AndLintDebug"
+                scheduled_lint = True
+            tasks.append(make_task(task, subtitle=subtitle + variant))
+
+        # Schedule "assemble and test" tasks; if a lintTask matches any variants and wasn't scheduled already, schedule it.
+        for variant in assemble_and_test:
+            task = as_module_task(module, "assemble" + variant) + " " + as_module_task(module, "test" + variant + "DebugUnitTest")
+            subtitle = "assembleAndTest"
+            if not scheduled_lint and variant in lint_task:
+                task = task + " " + as_module_task(module, lint_task)
+                subtitle = subtitle + "AndLintDebug"
+                scheduled_lint = True
+
+            tasks.append(make_task(task, subtitle=subtitle + variant))
+
+        # If we failed to find a matching assemble/assembleTest task to sneak lint into, schedule a separate one for it.
+        # Overhead of a separate task just for linting is quite significant, but this should be a rare case.
+        if not scheduled_lint:
+            lint_task = override.get("lintTask", "lintDebug")
+            tasks.append(make_task(as_module_task(module, lint_task), subtitle="onlyLintDebug"))
```

Comment:
tbh: Since the modules can have different source sets / overrides per release type I'd feel more comfortable running `lint` for all by default. However if that speeds up things noticeable then that may be fine. But since we are shipping `release` shouldn't we lint release instead of debug?

Reply:
Linting everything is about half of the runtime of the samples-browser task, so it is a big overhead. I'll switch this to `release`.